packages/core/src/interaction-output.ts

```diff
@@ -62,6 +62,8 @@ export class InteractionOutput implements WoT.InteractionOutput {
     }
 
     async arrayBuffer(): Promise<ArrayBuffer> {
+        // TODO check for this.schema == null also and return empty array?
```

Comment:
Question: Should we return an empty array for `arrayBuffer()` calls also, if there is no schema?

Reply:
Hmm, looking at the corresponding [algorithm](https://w3c.github.io/wot-scripting-api/#the-arraybuffer-function), I would say that `arrayBuffer()` should not take the schema into account at all (as it would also be the alternative method to use when the `value()` method fails).

Reply:
Yes I agree, however, I still think that the issue has to be solved in the specification (TD) first...

Reply:
> Yes I agree

Makes, sense, removed comment with https://github.com/eclipse-thingweb/node-wot/pull/1230/commits/0a9a69be9720c76f4f3af473ccfaf1a85eb0c1d2

> however, I still think that the issue has to be solved in the specification (TD) first...

I know, that there is something to fix/decide. Anyhow, I think this is still an improvement and fixes https://github.com/eclipse-thingweb/node-wot/issues/1228. Other/more changes can come later.
I let you decide...

Reply:
I might be ok for the sake of "let's fix this quickly" and then find a solution later. But can I ask you to add a warning log in the code and open an issue to keep this tracked?

Reply:
I created https://github.com/eclipse-thingweb/node-wot/issues/1243 to keep track of the _broader_ issue.

Anyhow, I am not sure what "warning log" I should/can add in the code. It is not really the fault of the user. It is up to **us** to fix the issues in TD/Scripting spec etc.
Do you have something specific in mind?

Reply:
I'd warn the user that he reading a value from a TD that does not have an explicit `schema` defined and we are treating it as if there is no output value.  There might not be what the TD designer wanted.

Reply:
In https://github.com/eclipse-thingweb/node-wot/pull/1230/commits/56133b995c0f9048a76a6af4d118793bed84341c I added a warn log. Please let me know whether that is sufficient.

Reply:
Comment resolved by now